File: src/main/java/org/motivepick/domain/entity/AbstractEntity.java
package org.motivepick.domain.entity

import javax.persistence.GeneratedValue
import javax.persistence.GenerationType.IDENTITY
import javax.persistence.Id
import javax.persistence.MappedSuperclass

@MappedSuperclass
abstract class AbstractEntity {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    var id: Long? = null
}

File: src/main/kotlin/org/motivepick/domain/entity/AbstractEntity.kt
package org.motivepick.domain.entity

import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType.IDENTITY
import jakarta.persistence.Id
import jakarta.persistence.MappedSuperclass

@MappedSuperclass
abstract class AbstractEntity {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    var id: Long = 0
}
